### PR TITLE
New tokens page

### PR DESCRIPTION
**Pull request by r-czajkowski**
Depends on: #2140 
Closes: #2003 

This PR updates the page where a user can delegate tokens. Also, it removes unnecessary components and improves some components by using a compound pattern(Dropdown, ProgressBar).

TODO:

- [x] update staking form
- [x] add dropdown on the `GrantedTokensPage`

obraz
obraz

obraz
obraz
obraz

### CODE REVIEW

**Comment by Shadowfiend** [comment]
Some quick takes here before I do code review:
- For grants, the “Withdraw Unlocked Tokens” section lists “Available to Withdraw”, but that number seems to be reflecting the total amount of the grant, not the amount that is currently withdrawable (roughly, unlocked - staked).
- I would love it if we very quickly started reworking cliffs that are currently listed as “a few seconds” to indicate “no cliff” instead.
- Tooltip icon spacing is off here, there should be a little more space between text and icon:
  Screen Shot 2020-11-24 at 11 16 20
- In FF, looks like the + icon for the max stake button is cut off:
  Screen Shot 2020-11-24 at 11 16 50
- The max stake button doesn't work for me :grimacing:
  Screen Shot 2020-11-24 at 11 22 44
- This may be more for @liz-shinn, but the combination of font size and color here is making for some nearly illegible text:
  Screen Shot 2020-11-24 at 11 17 30
- Our validation on form elements in the staking form are doing this annoying thing where if there's a validation error, the input shifts. Validation errors should appear in an existing space, not shift the input. The inputs also feel too tall for their contents, though that's probably a design note rather than a dev note.
- The grant allocation KEEP indicator (a) doesn't seem to show the precise number on hover and (b) has a hand cursor on hover. Same is true for the wallet balance, I believe.
- The ragged bottom here isn't great:  
  Screen Shot 2020-11-24 at 11 21 36
- @liz-shinn this is more of a general note, but we should consider listing a pending top-up that hasn't yet been committed at the top of the page, so the user is aware that they need to commit it.

With all of that said, this feels like an enormous improvement. Having the owned tokens up front especially is great. I think a fabulous additional step would be to hide the owned tokens entirely if your owned balance is 0, and vice versa for grants (though I realize grants might be harder, which is fine since they're not the landing tab).

**Review by Shadowfiend — COMMENTED** [review]
I think I maxed out on my review capacity for this PR heh. I'm not merging this quite yet as I want to make sure we've looked at TODOs and have a plan for the broken Max Stake button (whether it's removing or adding).

**Review by Shadowfiend — APPROVED** [review]
Actually I just realized #2160 is playing in this same playground, so let's merge this and handle any of those questions in #2160.

File: solidity/dashboard/src/components/DelegationOverview.jsx [hunk 1]
```diff
@@ -1,83 +1,41 @@
...
+  initializationPeriod,
+  keepTokenBalance,
+  grants = [],
+  selectedGrant = null,
+  context = "wallet",
+}) => {
   const cancelStakeSuccessCallback = useCallback(() => {
-    refreshData()
-  }, [refreshData])
+    // TODO
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply solidity/dashboard/src/components/DelegationOverview.jsx:27]: It's only a case when a user `cancel` stake from wallet tokens but I think we can handle it with an event subscription. I'll take a look.

File: solidity/dashboard/src/components/DelegationOverview.jsx [hunk 2]
```diff
@@ -87,65 +45,45 @@ const DelegationOverview = () => {
...
-              </span>
+            <Tag IconComponent={Icons.Time} text="Issued" />
+            <span className="ml-1">
+              {selectedGrant && formatDate(moment.unix(selectedGrant.start))}
             </span>
-          </>
-        )}
-      </div>
+          </div>
+        </header>
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply solidity/dashboard/src/components/DelegationOverview.jsx:68]: pls see https://github.com/keep-network/keep-core/pull/2113/commits/f75bfc7cfa920a033a63d6fe05f2434b46710d41

File: solidity/dashboard/src/components/DelegateStakeForm.jsx [hunk 3]
```diff
@@ -121,22 +125,35 @@ const TokensAmountField = ({
...
         </div>
-        <div className="text-caption text-grey-30">
-          Minimum stake is {displayAmount(minStake)} KEEP
-        </div>
       </div>
     </div>
   )
 }
 
+const TokenFieldAddon = ({ onClick }) => {
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply solidity/dashboard/src/components/DelegateStakeForm.jsx:153]: pls see https://github.com/keep-network/keep-core/pull/2113/commits/c9b3fe2ff2b6f6894e7af1e408a0ced668161426

File: solidity/dashboard/src/components/DelegationOverview.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Shadowfiend) [reply solidity/dashboard/src/components/DelegationOverview.jsx:68]: I ain't mad at that 💯

File: solidity/dashboard/src/components/TokenGrantOverview.jsx [hunk 4]
```diff
@@ -37,189 +30,146 @@ const TokenGrantOverview = ({ selectedGrant, selectedGrantStakedAmount }) => {
...
+          .from(moment.unix(selectedGrant.start), true)
+      : null
+  }, [selectedGrant.cliff, selectedGrant.start])
 
-export default TokenGrantOverview
+  const fullyUnlockedDate = useMemo(() => {
+    return selectedGrant.start && selectedGrant.duration
+      ? moment.unix(selectedGrant.start).add(selectedGrant.duration, "seconds")
+      : null
+  }, [selectedGrant.start, selectedGrant.duration])
```
[the remark this answers is not in the document]
Comment (Shadowfiend) [reply solidity/dashboard/src/components/TokenGrantOverview.jsx:46]: (Btw: not a blocker by any means, let's do it in follow up work.)

**Pull request merged**